carbon-design-system/carbon-for-ibm-dotcom
# fix(cta-block): fix medium breakpoint layout and spacing

### Related Ticket(s)

DDS issue: https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/7176
HC JIRA Ticket: https://jsw.ibm.com/browse/HC-2234

### Description

This PR fixes the medium breakpoint grid columns count and related spacing. It also removes a now-unneeded SCSS file.

### Testing Instructions

- Visit the `CTA Block` component 
  - Default: `?path=/story/components-cta-block--default`
  - With content items: `?path=/story/components-cta-block--with-content-items`
  - With link list: `?path=/story/components-cta-block--with-link-list`
- Check the layout and spacing at all breakpoints to make sure the component has the correct column count and spacing.
